landside/catkin_ws/src/gui/src/gui/launch_dialog.py

```diff
@@ -102,49 +117,200 @@ def node_name_selected(self, item_index):
+                                  f"is not in options list. It has been added as the first option.")
 
-                label = QtWidgets.QLabel(arg['name'])
-                input = QtWidgets.QLineEdit()
-                input.setText(default_value)
+        else:
+            rospy.logwarn(f"Options list for argument `{arg['name']}` in `{selected_node}` is not a "
+                          f"list of strings. Defaulting to unrestricted string input.")
 
-                # row inserted at position row+2, after the Package and Node Name rows
-                self.form_layout.insertRow(row + 2, label, input)
+        return input
 
-                arg['label'] = label
-                arg['lineEdit'] = input
+    def interpret_regex(self, selected_node, arg, doc_dict, input, toolTip):
+        try:
+            regex = QRegularExpression(doc_dict["regex"])
+            regex_validator = QtGui.QRegularExpressionValidator(regex)
+            if arg.get("default") is not None:
+                if regex.match(arg["default"]).hasMatch() or arg["default"] == "":
+                    input.setValidator(regex_validator)
+                    toolTip += f"Regex: {doc_dict['regex']}"
+                else:
+                    rospy.logwarn(f"Default value for argument `{arg['name']}` in `{selected_node}` "
+                                  f"does not match regex. Defaulting to unrestricted string input.")
+            else:
+                input.setValidator(regex_validator)
+                toolTip += f"Regex: {doc_dict['regex']}"
+
+        except Exception:
+            rospy.logwarn(f"Regex for argument `{arg['name']}` in `{selected_node}` is not valid. "
+                          f"Defaulting to unrestricted string input.")
+
+        return (input, toolTip)
+
+    def interpret_type(self, selected_node, arg, doc_dict, input, toolTip):
+        add_tooltip = True
+        if doc_dict["type"] == "bool":
+            if arg.get("default") is not None:
+                if arg["default"].lower() == "true":
+                    input = QtWidgets.QCheckBox()
+                    input.setChecked(True)
+                elif arg["default"].lower() == "false":
+                    input = QtWidgets.QCheckBox()
+                    input.setChecked(False)
+                else:
+                    rospy.logwarn(f"Default value for argument `{arg['name']}` in `{selected_node}` is "
+                                  f"not a valid boolean. Defaulting to unrestricted string input.")
+                    add_tooltip = False
+            else:
+                input = QtWidgets.QCheckBox()
+                input.setChecked(False)
+
+        elif doc_dict["type"] == "int":
+            int_validator = QtGui.QIntValidator()
+            if arg.get("default") is not None:
+                if (int_validator.validate(arg.get("default"), 0)[0] == QtGui.QValidator.Acceptable or
```

Comment:
Pretty sure you can just do `arg["default"]` here instead of `arg.get("default")`, now that you've checked that the default exists. Also, I would personally check the empty string case before the valid int case, since that's easier/faster to check.

Reply:
Removed all unnecessary `.get` operations from the file. Switched the order of the conditionals in the if statement.